packages/react-native-bpk-component-button/src/BpkButton.ios.js

```diff
@@ -20,100 +20,239 @@
+};
+
+const borderColors = {
+  disabled: colorGray100,
+  [BUTTON_TYPES.secondary]: colorGray100,
+  [BUTTON_TYPES.destructive]: colorGray100,
+};
+
+const gradientColors = {
+  [BUTTON_TYPES.primary]: { start: colorGreen500, end: colorGreen600 },
+  [BUTTON_TYPES.featured]: { start: colorPink500, end: colorPink600 },
+};
+
+const textColors = {
+  [BUTTON_TYPES.primary]: colorWhite,
+  [BUTTON_TYPES.featured]: colorWhite,
+  [BUTTON_TYPES.secondary]: colorBlue500,
+  [BUTTON_TYPES.destructive]: colorRed500,
+  disabled: colorGray300,
+};
+
+const getButtonComponent = (type: ButtonType) => {
+  if (type === BUTTON_TYPES.primary || type === BUTTON_TYPES.featured) {
+    return BpkGradientButton;
+  }
+  return BpkBorderedButton;
+};
+
+const getButtonColors = (
+  type: ButtonType,
+  themeAttributes: ?Theme,
+  disabled: boolean,
+) => {
+  if (type === BUTTON_TYPES.primary) {
+    if (disabled) {
+      return {
+        gradientStartColor: backgroundColors.disabled,
+        gradientEndColor: backgroundColors.disabled,
+      };
+    }
+    return themeAttributes
+      ? {
+          gradientStartColor: themeAttributes.buttonPrimaryGradientStartColor,
+          gradientEndColor: themeAttributes.buttonPrimaryGradientEndColor,
+        }
+      : {
+          gradientStartColor: gradientColors.primary.start,
+          gradientEndColor: gradientColors.primary.end,
+        };
+  }
+  if (type === BUTTON_TYPES.featured) {
+    if (disabled) {
+      return {
+        gradientStartColor: backgroundColors.disabled,
+        gradientEndColor: backgroundColors.disabled,
+      };
+    }
+    return themeAttributes
+      ? {
+          gradientStartColor: themeAttributes.buttonFeaturedGradientStartColor,
+          gradientEndColor: themeAttributes.buttonFeaturedGradientEndColor,
+        }
+      : {
+          gradientStartColor: gradientColors.featured.start,
+          gradientEndColor: gradientColors.featured.end,
+        };
+  }
+  if (type === BUTTON_TYPES.secondary) {
+    if (disabled) {
+      return {
+        backgroundColor: backgroundColors.disabled,
+        borderColor: borderColors.disabled,
+      };
+    }
+    return themeAttributes
+      ? {
+          backgroundColor: themeAttributes.buttonSecondaryBackgroundColor,
+          borderColor: themeAttributes.buttonSecondaryBorderColor,
+        }
+      : {
+          backgroundColor: backgroundColors.secondary,
+          borderColor: borderColors.secondary,
+        };
+  }
+  if (disabled) {
+    return {
+      backgroundColor: backgroundColors.disabled,
+      borderColor: borderColors.disabled,
+    };
+  }
+  return themeAttributes
+    ? {
+        backgroundColor: themeAttributes.buttonDestructiveBackgroundColor,
+        borderColor: themeAttributes.buttonDestructiveBorderColor,
+      }
+    : {
+        backgroundColor: backgroundColors.destructive,
+        borderColor: borderColors.destructive,
+      };
+};
+
+const getTextColor = (
```

Comment:
Can we move this and `textColors` to a single file? It looks exactly the same

Reply:
You're right, it is. I'll move it 👍